### DISCUSSION TITLE
Feature Request: Add passphrase support to the password generator

### DISCUSSION THREAD

**Discussion by JohnDoe-dev-spec** [pglombardo/PasswordPusher#472] [Ideas]
Hi,

I am self hosting the PasswordPusher and it is great, however i feel like the types of passwords it generates are quite hard to remember, is there a possibility to add a Passphrase generator to it so you can opt for random characters or a random passphrase instead?

Something like this; https://www.worksighted.com/random-passphrase-generator/#passphrase-generator

Apologises if its already been suggested, i've had a look and couldn't see anything obvious.

Thanks!

**Comment by pglombardo** [comment]
This is a good suggestion.  I'd like to revamp the password generator eventually.  I'm a bit underwater right now but when that happens, I'll try add words and passphrases.  Thanks for pointing this out.

**Reply by JohnDoe-dev-spec** [reply]
No problem, glad it was a useful suggestion. Hope everything else is ok though as you mentioned bit underwater.

Thanks again for your great work!

**Comment by jonnjonzzn** [comment]
This is the one reason we paused our roll out of pwpusher as the replacement for our previous solution.  And thank you ;-)

**Reply by pglombardo** [reply]
That's helpful info - thanks @jonnjonzzn.  I'll update here on the next update to the password generator.

**Comment by rickjames780** [comment]
This would be awesome if this could be possible. Id love for it to to be able to generate "Word-Word-##" formatted passwords.

**Comment by 0xab3d** [comment]
I use this custom generator.

https://www.gaijin.at/en/tools/password-generator?s=S&c=8&ps=bnwoQWEpKjR8bnxTfChBYSkqNnxTfChBYSkqNXxu&sc=LQ==&rc=MWxJb08w#result

**Comment by pglombardo** [comment]
Just a quick update.  With these settings (space as the only separator, no numbers, no titlecase) you can generate somewhat rememberable passphrases:

![Screenshot 2024-03-05 at 09 23 20](https://github.com/pglombardo/PasswordPusher/assets/395132/1c295c91-b935-4dee-82d0-7d5b06e90e82)

Eventually I will likely include word lists to generate passphrases from.  This is a good article from the EFF that I'll reference when I get back to this:

https://www.eff.org/dice
https://www.eff.org/deeplinks/2016/07/new-wordlists-random-passphrases

**Comment by ghostersk** [comment]
Bump, it would be nice to have there passphrase generator.
Users already hate long passwords, but making it something easy to read helps a lot.

**Comment by fourthfloorsolutions** [comment]
Just came to request this as well. We are looking to start using password pusher, but would really like this option. I use it a lot through Hudu, but would love to have this backed into password pusher

<img width="850" alt="Screenshot 2025-01-16 at 17 43 31" src="https://github.com/user-attachments/assets/6d4eed7e-e445-4592-ba62-4100143f357b" />

**Comment by pglombardo** [comment]
New password generator will be going out in the next release.  It has been merged to master and will go through final testing.

Pre-release is live here if you want to try it out: https://oss.pwpush.com/

This will be released for OSS first.  Pro/pwpush.com next.

Thanks all for the feedback & patience!

<img width="695" height="721" alt="Image" src="https://github.com/user-attachments/assets/7c8a59e5-f52e-469f-9d7e-fd2f67f5c34e" />
